views/default/elgg/popup.js

```diff
@@ -0,0 +1,172 @@
+define(function (require) {
+
+	var elgg = require('elgg');
+	var $ = require('jquery');
+	require('jquery-ui');
+
+	var popup = {
+		ready: false,
+		/**
+		 * Initializes a popup module
+		 * - Binds an event to hide visible popup modules on a click event outside of their DOM stack
+		 * - Binds an event to reposition the popup when window is scrolled or resized
+		 *
+		 * This method is called before the popup module is displayed
+		 * 
+		 * @returns {void}
+		 */
+		init: function () {
+			if (popup.ready) {
+				return;
+			}
+			$(document).on('click', function (e) {
+				var $eventTargets = $(e.target).parents().andSelf();
+				if ($eventTargets.is('[data-popup]')) {
+					return;
+				}
+				popup.close();
+			});
+
+			// Adding this magic so that popups with fixed position stick to their parent element
+			$(window).on('scroll resize', function () {
```

Comment:
I believe currently it remains in the viewport when you scroll, so it's a UX bug. I will see if I can migrate with minimal changes and then fluff it up in additional PRs